core/extension/gdextension_interface.h

```diff
@@ -2063,6 +2063,7 @@ typedef GDExtensionVariantPtr (*GDExtensionInterfaceArrayOperatorIndexConst)(GDE
 /**
  * @name array_ref
  * @since 4.1
+ * @deprecated in Godot 4.3. use `Array::operator=` instead.
```

Comment:
`Array::operator=` is however not accessible from GDExtension.
We only have a function that calls the copy constructor (obtained via `variant_get_ptr_constructor`):

https://github.com/godotengine/godot/blob/44e399ed5fa895f760b2995e59788bdb49782666/core/extension/gdextension_interface.h#L1225-L1236

So are we sure that the `array_ref` function has no use cases? Admittedly, such a method don't exist for most other variant types either...

Reply:
Okay, the wording might need to be updated, because it's not something that the GDExtension system needs to handle anyway. The only place where `_ref` was used before was for setting up typed arrays, which inherit from arrays

Unless I'm missing something, GDExtension doesn't need to access this directly. This is because the boilerplating for the array class it's extending from should be enough to make the connection in a dedicated typed_array script

Reply:
Testing some locally on `godot-cpp` & can confirm that its `typed_array.h` script is able to use `Array::operator=` instead of needing `_ref`